### DISCUSSION TITLE
Is it possible to export pure HTML files during static site generation?

### DISCUSSION THREAD

**Discussion by Mirodil** [vercel/next.js#11379] [Help]
Hello,

I could not find in the documentation how to rid of js file content during the static site build. Is it possible to export pure HTML files during static site generation?

Thank you,

**Comment by aralroca** [comment]
You can use a custom `_document.js` and don't render the scripts.

in the `_document`:

```jsx
{isFullyStaticPage ? <StaticHead /> : <Head />}
<body>
    <Main />
    {!isFullyStaticPage && <NextScript />}
</body>
```

And the **StaticHead** implementation:


```js
import { Head } from 'next/document'

/**
 * Remove all JavaScript from the document (convert to 100%
 * static page)
 */
export default class StaticHead extends Head {
  render() {
    const res = super.render()

    function transform(node) {
      // remove all link preloads
      if (
        node &&
        node.type === 'link' &&
        node.props &&
        node.props.rel === 'preload'
      ) {
        return null
      }
      if (node && node.props && node.props.children) {
        return {
          ...node,
          props: {
            ...node.props,
            children: Array.isArray(node.props.children)
              ? node.props.children.map(transform)
              : transform(node.props.children),
          },
        }
      }
      if (Array.isArray(node)) {
        return node.map(transform)
      }

      return node
    }

    return transform(res)
  }
}
```

**Comment by SachaG** [comment]
Interesting! I've used [this plugin](https://www.gatsbyjs.org/packages/gatsby-plugin-no-javascript/) to achieve the same thing in Gatsby, I think it can be worth exploring for static sites where performance is critical.

**Comment by jensmeindertsma** [comment]
Hello @Mirodil . Yes, this is possible and fully supported inside Next.js with a feature called `next export`. This exports all your pages to static HTML. [You can read more about it in the Next.js documentation](https://nextjs.org/docs/advanced-features/static-html-export)

**Reply by eric-burel** [reply]
Will it still work fine with js disabled in the browser for instance? I think that's the question here, and no-js behaviour is unclear in the doc

**Reply by jensmeindertsma** [reply]
I created an app with `create-next-app` and deployed to Vercel, and it works without JavaScript as you can try out for yourself [here](https://export-test.now.sh/). You are correct that there is barely any documentation about this, as this is not a recommended way of deploying your app. I couldn't think of a case in which the normal `next-build` is not superior. Only about 0.25% of users have JavaScript disabled and the entire UI still renders.

**Reply by timneutkens** [reply]
I landed an experimental option to disable runtime javascript @eric-burel @SachaG: https://github.com/zeit/next.js/pull/11949
